src/server/json_family.cc

```diff
@@ -1155,6 +1155,50 @@ OpResult<bool> OpSet(const OpArgs& op_args, string_view key, string_view path,
     std::move(*expr_result);                                                   \
   })
 
+void JsonFamily::MSet(CmdArgList args, ConnectionContext* cntx) {
+  vector<string_view> keys;
+  vector<string_view> paths;
+  vector<string_view> json_strs;
+  if (args.size() % 3 != 0) {
+    cntx->SendError(facade::WrongNumArgsError(cntx->cid->name()), facade::kSyntaxErrType);
+    return;
+  }
+
+  for (size_t i = 0; i < args.size(); i += 3) {
+    keys.emplace_back(ArgS(args, i));
+    paths.emplace_back(ArgS(args, i + 1));
+    json_strs.emplace_back(ArgS(args, i + 2));
+  }
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    OpResult<bool> res;
+    for (size_t i = 0; i < keys.size(); i += 3) {
+      string_view key = keys[i];
+      string_view path = paths[i + 1];
+      string_view json_str = paths[i + 2];
+      res = OpSet(t->GetOpArgs(shard), key, path, json_str, false, false);
```

Comment:
@romange, In case an individual operation of MSET fails, I'd like to undo right away the changes I've made and rewrite the keys with the copies that I stored before the changes. there is a way to lock the keys that I've already changed in this scope so the copies stay consistent?